exporters/otlp/src/otlp_populate_attribute_utils.cc

```diff
@@ -0,0 +1,245 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/exporters/otlp/otlp_populate_attribute_utils.h"
+
+namespace nostd = opentelemetry::nostd;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace otlp
+{
+
+//
+// See `attribute_value.h` for details.
+//
+const int kAttributeValueSize      = 16;
+const int kOwnedAttributeValueSize = 15;
+
+void OtlpPopulateAttributeUtils::PopulateAttribute(
+    opentelemetry::proto::common::v1::KeyValue *attribute,
+    nostd::string_view key,
+    const opentelemetry::common::AttributeValue &value) noexcept
+{
+  if (nullptr == attribute)
+  {
+    return;
+  }
+
+  // Assert size of variant to ensure that this method gets updated if the variant
+  // definition changes
+  static_assert(
+      nostd::variant_size<opentelemetry::common::AttributeValue>::value == kAttributeValueSize,
+      "AttributeValue contains unknown type");
+
+  attribute->set_key(key.data(), key.size());
+
+  if (nostd::holds_alternative<bool>(value))
```

Comment:
Can we use template to simplify this method?

Reply:
I don't think it would be possible, as we call different otlp method names for each type. Also, this is an existing function - I just copied it from one file to another to make it usable for metrics, traces, and logs :)